src/main/java/net/io/kino/service/impl/ShowroomServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package net.io.kino.service.impl;
+
+import net.io.kino.model.Showroom;
+import net.io.kino.repository.ShowroomRepository;
+import net.io.kino.service.ShowroomService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class ShowroomServiceImpl implements ShowroomService {
+
+    @Autowired
+    ShowroomRepository showrooms;
+
+    @Override
+    public Showroom createShowroom(Showroom showroom) {
+        List<Showroom> existingShowrooms = showrooms.findShowroomByName(showroom.getName());
+        if (existingShowrooms.size() > 0)
+        {
+            throw new IllegalArgumentException("This showroom already exists in repository.");
+        }
+        return showrooms.save(showroom);
+    }
+
+    @Override
+    public Showroom getShowroomById(long id){
+        return (Showroom) showrooms.findShowroomById(id);
```

Comment:
Kamieniowanie za castowanie `List<Showroom>` na Showroom

Reply:
Tutaj też można użyć findById żeby zwrócić jeden showroom, bo id jest unikalne to zawsze zwróci jeden element